### PR TITLE
[compiler-rt] fix __sanitizer::struct_sock_fprog_sz availability

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by devnexen** [comment]
@Zentrik would you be able to test this branch ?

**Comment by Zentrik** [comment]
I'm getting an error 
```
[09:12:16] /workspace/srcdir/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp:493:35: error: invalid application of ‘sizeof’ to incomplete type ‘__sanitizer::sock_fprog’
[09:12:16]   493 |   unsigned struct_sock_fprog_sz = sizeof(struct sock_fprog);
[09:12:16]       |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~
[09:12:16] In file included from /workspace/srcdir/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_posix.h:20,
[09:12:16]                  from /workspace/srcdir/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_libignore.cpp:16:
```

**Comment by devnexen** [comment]
nvm, just checked it s not available on musl, fixing ...

**Comment by devnexen** [comment]
> I'm getting an error
> 
> ```
> [09:12:16] /workspace/srcdir/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp:493:35: error: invalid application of ‘sizeof’ to incomplete type ‘__sanitizer::sock_fprog’
> [09:12:16]   493 |   unsigned struct_sock_fprog_sz = sizeof(struct sock_fprog);
> [09:12:16]       |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~
> [09:12:16] In file included from /workspace/srcdir/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_posix.h:20,
> [09:12:16]                  from /workspace/srcdir/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_libignore.cpp:16:
> ```

Can you give it a try when you get the chance ? thx.

File: compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc [hunk 1]
```diff
@@ -1313,7 +1313,7 @@ INTERCEPTOR(int, prctl, int option, unsigned long arg2, unsigned long arg3,
     COMMON_INTERCEPTOR_WRITE_RANGE(ctx, (u64 *)(arg5), sizeof(u64));
   } else if (res != -1 && option == PR_GET_PDEATHSIG) {
     COMMON_INTERCEPTOR_WRITE_RANGE(ctx, (u64 *)(arg2), sizeof(int));
-#  if !SANITIZER_ANDROID
+#  if !SANITIZER_ANDROID && !SANITIZER_MUSL
```
Comment (vitalybuka) [inline compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc:1316]: Why not just `#if SANITIZER_GLIBC` ?

Comment (vitalybuka) [reply compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc]: android can't be GLIBC

File: compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc [hunk 2]
```diff
@@ -1313,7 +1313,7 @@ INTERCEPTOR(int, prctl, int option, unsigned long arg2, unsigned long arg3,
     COMMON_INTERCEPTOR_WRITE_RANGE(ctx, (u64 *)(arg5), sizeof(u64));
   } else if (res != -1 && option == PR_GET_PDEATHSIG) {
     COMMON_INTERCEPTOR_WRITE_RANGE(ctx, (u64 *)(arg2), sizeof(int));
-#  if !SANITIZER_ANDROID
+#  if !SANITIZER_ANDROID && SANITIZER_GLIBC
```
Comment (vitalybuka) [inline compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc:1316]:
```suggestion
#  if SANITIZER_GLIBC
```

**Review by vitalybuka — APPROVED** [review]
With suggestion

**Comment by kostadinsh** [comment]
> > nvm, just checked it s not available on musl, fixing ...
> 
> It is available if you include the right kernel header (`linux/filter.h`, which for some reason is only included for glibc). Is there anything preventing that? (If not I can create a MR to fix this)

Looks like this comment was overlooked. Could you please submit the PR anyway and see what they say?

**Comment by maribu** [comment]
https://github.com/llvm/llvm-project/pull/183411

**Pull request merged**